### PR TITLE
Added radius rescaling to simple env

### LINKED ISSUE #1188: [Bug Report] Strange behavior in MPE Simple-Tag rendering
### Describe the bug

I'm using the Simple Tag scenario to run some MARL algorithms. However, when I used the "human" or the "rgb_array" render mode to visualize the trained policies interacting, the animation seemed very odd to me. It does not seem to be behaving as described on the MPE docs page.

From the Render section in the MPE docs, "Rendering displays the scene in a window that automatically grows if agents wander beyond its border". I do not see this happening in my results.

Also, when I look at the generated animation, it seems the landmarks are actually moving. I've even confirmed that the landmarks are not moving by checking their positions and velocities from the unwrapped env, just to be sure (in the code below):

```
Landmark position [-0.58032043  0.22282363]
Landmark velocity [0. 0.]

Landmark position [-0.32183733 -0.8377423 ]
Landmark velocity [0. 0.]

Landmark position [-0.58032043  0.22282363]
Landmark velocity [0. 0.]

Landmark position [-0.32183733 -0.8377423 ]
Landmark velocity [0. 0.]

Landmark position [-0.58032043  0.22282363]
Landmark velocity [0. 0.]

Landmark position [-0.32183733 -0.8377423 ]
Landmark velocity [0. 0.]
```

Here's a GIF generated with the code below using the standard loop with random agents:

simple_tag

### Code example

```shell
from pettingzoo.mpe import simple_tag_v3
from matplotlib import pyplot as plt
from PIL import Image


env = simple_tag_v3.parallel_env(render_mode="rgb_array", max_cycles=25)

observations, infos = env.reset()
animation = []
while env.agents:
    # this is where you would insert your policy
    actions = {agent: env.action_space(agent).sample() for agent in env.agents}

    observations, rewards, terminations, truncations, infos = env.step(actions)
    for landmark in env.unwrapped.world.landmarks:
        print(f'Landmark position {landmark.state.p_pos}')
        print(f'Landmark velocity {landmark.state.p_vel}')
    
    image = env.render()
    animation.append(Image.fromarray(image))
    plt.imshow(image)

env.close()

image_file = './simple_tag.gif'
animation[0].save(
    image_file, save_all=True, append_images=animation[1:], loop=0, duration=1)
```

### System info

**Installation**
pip install pettingzoo[mpe]

**Versions**
PettingZoo version: 1.24.3
PyGame version: 2.3.0

**OS**
Ubuntu 22.04.4 LTS

### Additional context

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/PettingZoo/issues) in the repo

### PR DESCRIPTION
**Pull request by pandyah5**
# Description

This PR implements dynamic rescaling of the entity radius as we 'Zoom In' or 'Zoom out' of the rendering. This fixes #1188 which mentions that keeping the radius constant when rescaling the coordinates leads to an illusion of fixed obstacles coming closer or moving away from each other when in reality they are fixed.

Files modified: simple_env.py
High-level overview: Extended the concept of cam_range and recorded the original cam_range to scale the radius

Fixes #1188 , Depends on None

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update [Mentioning this in the documentation may be helpful]

### Screenshots

**Notice how the zoom out effect is more intuitive in the dynamic resizing case**

Before dynamic resizing:
before

After dynamic resizing:
after

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no new errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by pandyah5** [comment]
Hi @elliottower, just wanted to bring your attention to this PR. Let me know if it requires any changes, would be glad to add them!

File: pettingzoo/mpe/_mpe_utils/simple_env.py
Comment (jjshoots) [inline pettingzoo/mpe/_mpe_utils/simple_env.py:15]: Apologies for necro-ing this PR. The devs for PettingZoo recently went on a short hiatus.

Could this not be a global variable and be an environment argument instead? Also default it to True so we don't break existing learned algorithms which utilize image space.

Comment (jjshoots) [inline pettingzoo/mpe/_mpe_utils/simple_env.py:15]: Could this not be a global variable and be an environment argument instead? Also default it to True so we don't break existing learned algorithms which utilize image space.

**Review by jjshoots — CHANGES_REQUESTED** [review]
Apologies for necro-ing this PR. the maintainers of PZ went on a short hiatus and are now getting back to things.

File: pettingzoo/mpe/_mpe_utils/simple_env.py
Comment (pandyah5) [reply pettingzoo/mpe/_mpe_utils/simple_env.py]: Totally understandable, everyone deserves some time off! I have made it into an env argument and changed the default to False (I guess that's what you meant instead of True). 

Note: It required changes in a lot of places including documentation hence you'll see multiple files in the PR.

Comment (jjshoots) [reply pettingzoo/mpe/_mpe_utils/simple_env.py]: Thanks for the help!

**Comment by jjshoots** [comment]
Thanks for the fix! Looks good to go for me, once precommit is fixed then we can merge. :)

**Comment by pandyah5** [comment]
> Thanks for the fix! Looks good to go for me, once precommit is fixed then we can merge. :)

Gotcha, for the precommit I just need to modify the formatting of my files right? The tutorial failures seem unrelated to this PR.

**Comment by pandyah5** [comment]
@jjshoots I have modified the formatting and the pre-commit check passes now. Thanks for reviewing my PR!

PS: I just glanced over your website, your journey across the 'remote Pacific' is sooo cool. Is there a place where I can message you to learn more about it? 😄

**Comment by jjshoots** [comment]
LGTM! Thanks for the addition.

You can contact me on Discord @jjshoots

**Review by jjshoots — APPROVED** [review]

**Pull request merged**
